apis/validator/duties/sync.yaml

```diff
@@ -0,0 +1,68 @@
+post:
+  tags:
+    - ValidatorRequiredApi
+    - Validator
+  summary: "Get sync committee duties"
+  operationId: "getSyncCommitteeDuties"
+  description: "Requests the beacon node to provide a set of sync committee duties for a particular epoch.
+
+    Duties should only need to be checked once per epoch,
+    however a chain reorganization (of > MIN_SEED_LOOKAHEAD epochs) could occur,
+    resulting in a change of duties. For full safety, you should monitor head events and confirm the
+    dependent root in this response matches:
+
+      - event.previous_duty_dependent_root when `compute_epoch_at_slot(event.slot) == epoch`
+
+      - event.current_duty_dependent_root when `compute_epoch_at_slot(event.slot) + 1 == epoch`
+
+      - event.block otherwise
+
+    The dependent_root value is `get_block_root_at_slot(state, compute_start_slot_at_epoch(epoch - 1) - 1)`
+    or the genesis block root in the case of underflow."
```

Comment:
This won't be the right dependent root for sync committees as they span multiple epochs.  I think it winds up being:

```
previous_sync_commitee_period = epoch / EPOCHS_PER_SYNC_COMMITTEE_PERIOD - 1
first_epoch = previous_sync_committee_period * EPOCHS_PER_SYNC_COMMITTEE_PERIOD

get_block_root_at_slot(state, compute_start_slot_at_epoch(first_epoch) - 1)
```
again with genesis block in the case of underflow.

In other words it should be the last block of the epoch prior to the start of the previous sync committee period. That however is a bit of a problem because individual block root history only goes back 256 epochs which is a single sync committee period.  So you can't use `get_block_root_at_slot` to get the dependent root in this case and would actually need a quite old state.

Notably the head event doesn't include that value so it would also have to have a current and previous sync committee dependent root field added for this to be useful.

Given the exceptionally long re-org that would be required to change sync committee assignments I wonder if it's worth including the dependent root at all.

Reply:
> Given the exceptionally long re-org that would be required to change sync committee assignments I wonder if it's worth including the dependent root at all.

I agree.  When I wrote the original API spec the sync committee was based on the current epoch so this would have been required, but if we now have this much lookahead I don't see the dependent root being relevant.

Reply:
ok I'll remove it from the response.